sdk/monitor/monitor-query/src/internal/modelConverters.ts

```diff
@@ -122,14 +121,45 @@ export function convertResponseForQueryBatch(
         return left - right;
       })
       ?.map((response: GeneratedBatchQueryResponse) => ({
-        id: response.id,
         status: response.status,
+        visualization: response.body?.render,
+        statistics: response.body?.statistics,
         // hoist fields from the sub-object 'body' to this level
         error: response.body?.error,
-        tables: response.body?.tables?.map(convertGeneratedTable)
-      }))
+        tables: response.body?.tables?.map((table) => convertGeneratedTable(table)),
+        logsQueryResultStatus: "Success"
+      })),
+    batchResultStatus: "AllSucceeded"
   };
-
+  // compute status for failed or succeed or partial results
+  if (newResponse.results) {
+    let resultStatus = "Success";
+    let errorCount: number = 0;
+    for (let i = 0; i < newResponse.results?.length; i++) {
+      if (newResponse.results[i].error && newResponse.results[i].tables) {
+        newResponse.results[i].logsQueryResultStatus = "Partial";
+        resultStatus = "Partial";
+      } else if (newResponse.results[i].tables) {
+        newResponse.results[i].logsQueryResultStatus = "Success";
+      } else {
+        newResponse.results[i].logsQueryResultStatus = "Failed";
+        errorCount++;
+      }
+    }
+    if (errorCount === newResponse.results?.length - 1) {
+      newResponse.batchResultStatus = "AllFailed";
```

Comment:
I was surprised to see that these result statuses aren't defined as constants somewhere. Is that something the JS SDKs decided not to do?

Reply:
I removed the batchResultStatus flag.